packages/quick_actions/quick_actions/ios/Classes/FLTQuickActionsPlugin.m

```diff
@@ -50,12 +52,36 @@ - (BOOL)application:(UIApplication *)application
     performActionForShortcutItem:(UIApplicationShortcutItem *)shortcutItem
                completionHandler:(void (^)(BOOL succeeded))completionHandler
     API_AVAILABLE(ios(9.0)) {
-  [self.channel invokeMethod:@"launch" arguments:shortcutItem.type];
+  [self _handleShortcut:shortcutItem.type];
   return YES;
 }
 
+- (BOOL)application:(UIApplication *)application
+    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  if (@available(iOS 9.0, *)) {
+    UIApplicationShortcutItem *shortcutItem =
```

Comment:
Why don't we  directly call `handleShortcut` here?

Reply:
We have to wait here for the application to become fully active before the method channel are fully initialised. When we directly call the `handleShortcut` in the `didFinishLaunchingWithOptions` method the channels on the Dart side are not initialised yet and the `invokeMethod:@"launch"` isn't processed.

Reply:
Ah makes sense. Let's add a comment here for future reference.

Why do we return `No` below?

Reply:
According to Apple's documentation you should return `NO` if you already handled the shortcut action which will make sure the `application:performActionFor:` method and not handle the same action again.

The explanation can be found in the last alinea of the "Discussion" section of this article: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623032-application

Reply:
OK. Let's also add comment here too. Then I think we are good to go!

Reply:
Thanks @cyanglaz, I have added comments as explanation.